plugins/text_generation_webui_plugin.py

```diff
@@ -50,22 +55,22 @@ def streaming_request_http(self, query, user_id):
             "stream": True,
         }
 
-        chunks = []
-        ack_time = 0
-        first_token_time = 0
-        start_time = time.time()
+        result = RequestResult(user_id, query.get("text"), query.get("input_tokens"))
+
+        tokens = []
+        result.start_time = time.time()
 
-        # TODO add configurable timeout to requests
         response = requests.post(
-            self.route, headers=headers, json=data, verify=False, stream=True
+            self.host, headers=headers, json=data, verify=False, stream=True
```

Comment:
Why is there `stream=True` in the post method as well as in the data payload above `stream: True` as per the Openai API [documentation](https://github.com/openai/openai-openapi/blob/master/openapi.yaml) we need to enable streaming in data payload but is it also required in the `request.post` method?

Reply:
I believe both are needed. I guess that the stream=True in the payload is needed to tell the Server to respond with a  stream, and the stream in the requests.post method is used to tell the requests client to receive the response as a stream. Other APIs like TGIS have different endpoints for streaming vs non-streaming (`/generate` vs `/generate_stream`)